### DISCUSSION TITLE
7.2.3 (New profile logic, performance tuning)

### DISCUSSION THREAD

**Discussion by T-Troll** [T-Troll/alienfx-tools#204] [Announcements]
###  Changes into this release:

`All fan control apps`:
- CPU usage decreased for most fan control operation (about 40% decrease). There are still some points to decrease it even more.
- GPU boost removed completely from code (it's impossible to do it via WMI).

`All light-control apps`:
- SDK device detection and initialization optimization (now about 4 times faster).
- 2nd color for -lock keys (Caps lock, Num lock) for APIv8 devices (external keyboards). One light can have more, than one ID (ID 6 in this case for "on" color).

`AlienFX-GUI`:
- Profile handling re-worked:
  - AC/DC trigger work as default profile and at auto-switch to profile:
    - In case you have a profile with no application, but selected AC/DC trigger, app with switch to it (instead of default one) if current system power source meets it.
    - For auto switch profile selection, current power source also into the game - it first matches profile with selected power source, then with not selected.
  - Profile selected by key press block profile switch until key released.
  - Click  "Remove" or "Clean" ("-") button with "Shift" key pressed to suppress confirmation dialogue.
  - Bug fixed: Profile switch by key does not react for some keys.
  - Bug fixed: Exception at profile deletion.
- New tab selection logic if no lights or fans detected - other tabs still available for use.
- Status chain switch optimization (faster app start, faster reaction to profile/effect/tab switch).
- Bug fixed: Tray icon tooltip not updated if "Check for updates" off.
- Bug fixed: Exception, then select grid and have lights from disconnected device into it.
- Key-switched grid effect performance optimization

<hr /><em>This discussion was created from the release <a href='https://github.com/T-Troll/alienfx-tools/releases/tag/7.2.3'>7.2.3 (New profile logic, performance tuning)</a>.</em>

**Comment by Rixtho** [comment]
After upgrading to 7.2.3 from 7.0.0 none of the fan profiles match my manual setting anymore. And the fan profiles are also somehow messed up (see image below). Did anyone else also have this issue?
![image](https://user-images.githubusercontent.com/74471193/192150149-78f424cf-544c-4fc7-9f01-8b991c5000e3.png)

**Reply by T-Troll** [reply]
It was a bug after resume from sleep.
Check latest release (7.3.0), should work correctly now.

**Comment by T-Troll** [comment]
Yes, looks incorrect, let me check and maybe provide you debug version.
For the beginning, can you please run reedit and export `HKCU/Software/Alienfxgui` for me (it can be a glitch into profile settings, for example)?

**Reply by Rixtho** [reply]
Sorry for the late response but here it is.
[Alienfxgui.zip](https://github.com/T-Troll/alienfx-tools/files/9674027/Alienfxgui.zip)
